docs/new-architecture-backward-turbomodule.md

```diff
@@ -0,0 +1,699 @@
+    "!ios/build",
+    "!**/__tests__",
+    "!**/__fixtures__",
+    "!**/__mocks__"
+  ],
+  "keywords": ["react-native", "ios", "android"],
+  "repository": "https://github.com/<your_github_handle>/example-library",
+  "author": "<Your Name> <your_email@your_provide.com> (https://github.com/<your_github_handle>)",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/<your_github_handle>/example-library/issues"
+  },
+  "homepage": "https://github.com/<your_github_handle>/example-library#readme",
+  "publishConfig": {
+    "registry": "https://registry.npmjs.org/"
+  },
+  "devDependencies": {},
+  "peerDependencies": {
+    "react": "*",
+    "react-native": "*"
+  }
+}
+```
+
+For sake of simplicity, we will go on with this manual setup, adding files while we work.
+Remember to update the `<Your Name>`, `<your_email@your_provide.com>`, and all the `<your_github_handle>` tokens, if you want to follow this approach with your information.
+
+### Create a Testing Environment
+
+To develop a Native Module we would like to test the JS side and to develop safely the native parts. A library usually does not come with a testing aplication out-of-the-box. So, we are going to need one.
+
+:::note
+If you used the React Native CLI to create the module, you have an application called `example` within you `example-library` folder.
+:::
+
+To create a proper test environment, let's navigate to the parent folder of our `example-library` and run the following commands:
+
+```sh
+npx react-native init OldArchitectureApp
+cd OldArchitectureApp
+yarn install
+```
+
+Now, open a new terminal and start Metro by running:
+
+```sh
+npx react-native start
+```
+
+Go back to previous terminal and run:
+
+```sh
+npx react-native run-ios
+```
+
+When all the processes are done, you should see the default React Native application, running on a simulator.
+We will update the code of this app to test our Native Module.
+
+### Define the JS Shared Code
+
+The next step is to define how the React Native app will use our module. Native modules are registered using the `NativeModule` object provided by the React Native framework.
+
+For this example, we are going to create a manager that asks the GPS authorization to the user.
+
+As first step, let's create a folder named `src` and an `index.js` file into it. Add the following code into the JS file.
+
+```js title="index.js"
+// @flow
+// Import the required objects
+import { NativeModules, Platform } from 'react-native';
+
+// Declare an error message, triggered when the Native Module is used but not properly linked.
+const LINKING_ERROR =
+  'The package "AuthorizationManager" doesn\'t seem to be linked. Make sure: \n\n' +
+  Platform.select({
+    ios: "- You have run 'pod install'\n",
+    default: ''
+  }) +
+  '- You rebuilt the app after installing the package\n' +
+  '- You are not using Expo managed workflow\n';
+
+// Create a Proxy object that raises the error if the module is not linked.
```

Comment:
Is the proxy object needed?

Reply:
Not really. It is something create by template that I find helpful to get some clear error message.
I agree, it makes things more complex. I'll remove it.